storm/src/main/storm/mesos/MesosCommon.java

```diff
@@ -72,8 +72,15 @@ public static String getMesosComponentNameDelimiter(Map conf, TopologyDetails in
         .or(DEFAULT_MESOS_COMPONENT_NAME_DELIMITER);
   }
 
+  public static String timestampMillis() {
```

Comment:
I feel this is a bit _more_ obfuscated than the existing logic.  At the minimum I would break them out like this:

```
String now = String.valueOf(System.currentTimeMillis());
String secs = now.substring(0, now.length()-3);
String msecs = now.substring(now.length()-3, now.length());
return String.format("%s.%s", secs, msecs);
```

Is that better than the / and %?  Maybe it is?  `¯\_(ツ)_/¯`